iocore/net/BIO_fastopen.cc

```diff
@@ -27,6 +27,67 @@
 
 #include "BIO_fastopen.h"
 
+// For BoringSSL, which for some reason doesn't have this function.
+// (In BoringSSL, sock_read() and sock_write() use the internal
+// bio_fd_non_fatal_error() instead.) #1437
+//
+// The following is copied from
+// https://github.com/openssl/openssl/blob/3befffa39dbaf2688d823fcf2bdfc07d2487be48/crypto/bio/bss_sock.c
```

Comment:
That works ... Is the purpose to avoid having to attribute it?

In general, I like to code for OpenSSL, and add fixes to make other environments conform to it -- unless there's a reason not to. So in this case I'd continue calling BIO_sock_non_fatal_error(), and copy it verbatim from OpenSSL if it's missing.

Functionally, your solution is equivalent (and cleaner!), but unless there's a reason not to use OpenSSL's function, that's the style I prefer. I'm happy either way, though :-)

I'll open a PR to add attribution to the NOTICE file.